plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2463,6 +2482,7 @@ public ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session,
                 hiveTable.getAnalyzePartitionValues(),
                 hiveTable.getAnalyzeColumnNames(),
                 Optional.empty(),
+                Optional.empty(), // Projected columns is used only during optimization phase of planning
```

Comment:
method doc is:
```
     * Return a table handle whose partitioning is converted to the provided partitioning handle,
     * but otherwise identical to the provided table handle.
```
so you should be able to just pass `tableHandle` projections

Reply:
I know it would be valid to pass the tableHandle projections. I just avoided it for the sake of efficiency.
It appears that we've done the same for `constraintColumns` as well.